homeassistant/components/sun.py

```diff
@@ -139,11 +142,13 @@ def __init__(self, hass, location):
         self.hass = hass
         self.location = location
         self._state = self.next_rising = self.next_setting = None
+        track_utc_time_change(hass, self.timer_update, second=30)
 
     @property
     def should_poll(self):
-        """ We trigger updates ourselves after sunset/sunrise """
-        return False
+        """ We trigger updates ourselves after sunset/sunrise,
+            but sun angle requires poll """
+        return True
```

Comment:
This property is pretty bogus to begin with as we're not using the entity component which means that no one will check this value. Since we do however trigger our own track state changes, it is better suited at `False`.